docs/src/modules/spring/pages/actions-publishing-subscribing.adoc

```diff
@@ -1 +1,117 @@
-= Publishing and Subscribing
+= Publishing and Subscribing with Actions
+
+include::ROOT:partial$include.adoc[]
+
+A very common use case when building Microservices is to publish and subscribe to a stream of events. The source of events can be the journal of an event sourced entity, the value entity state changes, a https://cloud.google.com/pubsub/docs/overview[Google Cloud Pub/Sub] or https://kafka.apache.org/[Apache Kafka] topic for asynchronous messaging between services.
+
+With Actions you can:
+
+- subscribe to events emitted by an event sourced entity within the same service.
+- subscribe to state changes emitted by a value entity within the same service.
+- subscribe to events from Event Sourced Entities published as xref:service-to-service.adoc[service to service eventing]
+- subscribe to external events from Google Cloud Pub/Sub or Apache Kafka.
+- publish events to a Google Cloud Pub/Sub or Apache Kafka topic.
+
+Messages are guaranteed to be delivered at least once. This means that receivers must be able to handle duplicate messages.
+
+== Subscribing to an Event Sourced Entity
+
+You can subscribe an Action to events from an Event Source Entity. To receive messages from the entity, you can annotate a method with `@Subscribe.EventSourcedEntity` and specify the class of the entity. The input type of the method must be the same of the events generated by the entity.
+
+The following example assumes the existence of an xref:eventsourced.adoc[Event Sourced Entity] counter that emits events of type `CounterEvent` with subtypes `ValueIncreased` and `ValueDecreased`.
+
+[source,java,indent=0]
+.src/main/java/com/example/actions/SubscribeAction.java
+----
+include::example$spring-eventsourced-counter/src/main/java/com/example/actions/SubscribeAction.java[tags=class;sub-ESE-action]
+----
+<1> subscribing to the events from the Counter.
+<2> setting the method input type to the events produced by the counter.
+<3> any return is valid. See NOTE.
+
+The events from the entity are delivered to the Action. The implementation may vary, for this simplified example you are just logging it, but it could a forward to some other component or external service.
+
+[NOTE]
+====
+The return value of the method is an `Action.Effect` with message `0`, but can be any other of type `Action.Effect<Integer>` as the return type of the method indicates. The Kalix framework needs the response to ensure that the event was successfully processed. If no exception is thrown and the method returns a value, the framework assumes that the event was successfully processed and marks it as such. This allows the next event to be sent to the subscribing method.  However, if an exception is raised and not handled, this action will not process any more events until the necessary handling is added. It will raise the same error over and over again until the application is fixed and restarted.
+====
```

Comment:
I think is important to explain that to the user even if we return `Confirmed()`. Otherwise, the user will have the wrong model of how this works. It is very useful for me to have this model in mind and I remember when I found out that it was an aha moment. 
I'm happy to introduce a record  like`Confirmed` if you think is better. Please, let me know.

Reply:
Good to explain it. Good idea. 

We can convert this NOTE into an explanation about the meaning of returning `effects().reply(Confirmed.instance)` or returning `effects().error("something went wrong")` or even throwing an exception.